zuul-core/src/main/java/com/netflix/zuul/netty/server/ClientRequestReceiver.java

```diff
@@ -254,6 +255,7 @@ private HttpRequestMessage buildZuulHttpRequest(final HttpRequest nativeRequest,
         // This is the only way I found to get the port of the request with netty...
         final int port = channel.attr(SourceAddressChannelHandler.ATTR_SERVER_LOCAL_PORT).get();
         final String serverName = channel.attr(SourceAddressChannelHandler.ATTR_SERVER_LOCAL_ADDRESS).get();
+        final SocketAddress clientDestinationAddress = channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_ADDR).get();
```

Comment:
Is this attribute guaranteed to be set?

Reply:
Not really, but the getter's API now indicates an optional value, which should be sufficient.